app/src/main/java/com/github/odaridavid/weatherapp/data/weather/RefreshWeatherUseCase.kt

```diff
@@ -0,0 +1,45 @@
+package com.github.odaridavid.weatherapp.data.weather
+
+import android.content.Context
+import com.github.odaridavid.weatherapp.R
+import com.github.odaridavid.weatherapp.core.api.WeatherRepository
+import com.github.odaridavid.weatherapp.core.model.DefaultLocation
+import com.github.odaridavid.weatherapp.core.model.Weather
+import com.github.odaridavid.weatherapp.util.NotificationUtil
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+import javax.inject.Inject
+
+class RefreshWeatherUseCase @Inject constructor(
```

Comment:
Nice work on the usecase, 

Preferably it would implement an interface i.e. `RefreshWeatherUseCase` that is part of the `core` package to represent business domain use cases, it's also the pattern I intended to use on other parts as functionality increases.

Reply:
updated. Kindly check